sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,158 @@
+    assert A.dcm(B) == Matrix([[cos(q2), -sin(q2), 0], [sin(q2), cos(q2), 0], [0, 0, 1]])
+    assert N.dcm(B).simplify() == Matrix([[cos(q1 + q2), -sin(q1 + q2), 0], \
+        [sin(q1 + q2), cos(q1 + q2), 0], [0, 0, 1]])
+
+    #Check Angular Velocity
+    assert A.ang_vel_in(N) == u1 * N.z
+    assert B.ang_vel_in(A) == u2 * A.z
+    assert B.ang_vel_in(N) == u1 * N.z + u2 * A.z
+
+    #Check kde
+    assert J1.kdes() == [u1 - q1.diff(t)]
+    assert J2.kdes() == [u2 - q2.diff(t)]
+
+    #Check Linear Velocity
+    assert PartP.masscenter.vel(N) == l*u1*A.y
+    assert PartR.masscenter.vel(A) == l*u2*B.y
+
+def test_pin_joint_chaos_pendulum():
+    mA, mB, lA, lB, h = symbols('mA, mB, lA, lB, h')
+    theta, phi, omega, alpha = dynamicsymbols('theta phi omega alpha')
+    N = ReferenceFrame('N')
+    A = ReferenceFrame('A')
+    B = ReferenceFrame('B')
+    lA = (lB - h / 2) / 2
+    lC = (lB/2 + h/4)
+    rod = Body('rod', frame=A, mass=mA)
+    plate = Body('plate', mass=mB, frame=B)
+    C = Body('C', frame=N)
+    J1 = PinJoint('J1', C, rod, coordinates=theta, speeds=omega, child_joint_pos=lA*A.z,\
+        parent_axis=N.y, child_axis=A.y)
+    J2 = PinJoint('J2',rod, plate, coordinates=phi, speeds=alpha, parent_joint_pos=lC*A.z,\
+        parent_axis=A.z, child_axis=B.z)
+
+    #Check orientation
+    assert A.dcm(N) == Matrix([[cos(theta), 0, -sin(theta)], [0, 1, 0], [sin(theta), 0, cos(theta)]])
+    assert A.dcm(B) == Matrix([[cos(phi), -sin(phi), 0], [sin(phi), cos(phi), 0], [0, 0, 1]])
+    assert B.dcm(N) == Matrix([[cos(phi)*cos(theta), sin(phi), -sin(theta)*cos(phi)], \
+        [-sin(phi)*cos(theta), cos(phi), sin(phi)*sin(theta)], [sin(theta), 0, cos(theta)]])
+
+    #Check Angular Velocity
+    assert A.ang_vel_in(N) == omega * N.y
+    assert A.ang_vel_in(B) == - alpha * A.z
+    assert N.ang_vel_in(B) == - omega * N.y - alpha * A.z
+
+    #Check kde
+    assert J1.kdes() == [omega - theta.diff(t)]
+    assert J2.kdes() == [alpha - phi.diff(t)]
+
+    #Check pos of masscenters
+    assert C.masscenter.pos_from(rod.masscenter) == lA*A.z
+    assert rod.masscenter.pos_from(plate.masscenter) == - lC * A.z
+
+    #Check Linear Velocities
+    assert rod.masscenter.vel(N) == (h/4 - lB/2)*omega*A.x
+    assert plate.masscenter.vel(N) == ((h/4 - lB/2)*omega + (h/4 + lB/2)*omega)*A.x
+
+def test_pinjoint_arbitrary_axis():
+    theta, omega = dynamicsymbols('J_theta, J_omega')
+    N = ReferenceFrame('N')
+    A = ReferenceFrame('A')
+    P = Body('P', frame=N)
+    C = Body('C', frame=A)
+    PinJoint('J', P, C, parent_axis=N.y, parent_joint_pos=N.z, child_joint_pos=A.z)
+
+    axis1 = N.y.normalize()
+    axis2 = A.x.normalize()
+    angle = acos(dot(axis2, axis1))
+
+    assert angle == 0 #Axis are aligned
+    assert A.dcm(N) == Matrix([[0, 1, 0], [-cos(theta), 0, sin(theta)], [sin(theta), 0, cos(theta)]])
+    assert A.ang_vel_in(N) == omega * N.y
+    assert C.masscenter.vel(N) == omega * N.x + omega * A.y
```

Comment:
@moorepants I think the velocity for this case that we calculated in meeting yesterday wasn't correct as the distance between the `P.masscenter` and `C.masscenter` is defined a unit vector in each frame, so I think the velocity wouldn't be defined in just 1 frame.

Reply:
If we remove either one of the joint position the result's same as it is in the case when axis are aligned, which kind of makes me feel this result is correct. Can you please crosscheck while I'm trying to learn how to calculate it manually.

Reply:
The dcms seems correct to me.